### PR TITLE
[Bugfix] [Tiered Caching] Make ehcache disk cache size setting be ByteSizeValue rather than long

### PR DESCRIPTION
**Pull request by peteralfonsi**
### Description
EhcacheDiskCache's DISK_CACHE_MAX_SIZE_IN_BYTES_SETTING should take ByteSizeValue, not long. This means users can specify values with units like "10G", "10GB", or "10737418240B" when passing settings in the command line, rather than having to pass "10737418240". This matches the behavior of similar settings like OpenSearchOnHeapCacheSettings.MAXIMUM_SIZE_IN_BYTES. 

Added unit test around this change. Also manually tested with the IntelliJ debugger and commands like: 
```./gradlew run -Dtests.opensearch.opensearch.experimental.feature.pluggable.caching.enabled=true -Dtests.opensearch.indices.requests.cache.store.name=tiered_spillover -Dtests.opensearch.indices.requests.cache.tiered_spillover.onheap.store.name=opensearch_onheap -Dtests.opensearch.indices.requests.cache.tiered_spillover.disk.store.name=ehcache_disk -PinstalledPlugins="['cache-ehcache']" -Dtests.opensearch.indices.requests.cache.ehcache_disk.storage.path="/Volumes/workplace/opensearch/OpenSearch/build/testclusters/runTask-0/data/nodes/0/indices/request_cache" -Dtests.opensearch.indices.requests.cache.ehcache_disk.max_size_in_bytes=10M```

to ensure the actual sizes received by ehcache are correct in all cases. 

### Related Issues
Part of [tiered caching](https://github.com/opensearch-project/OpenSearch/issues/10024). 

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
~- [N/A] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).~
- [x] Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))
- [x] Commits are signed per the DCO using --signoff
~- [N/A] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))~
~- [N/A] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by jainankitk — APPROVED** [review]
I think this change should be backward compatible since the new nodes should be able to understand both. Approved!

**Comment by jainankitk** [comment]
@peteralfonsi @sgup432 - Are we planning to merge this change? If yes, let us address the failing checks and get approval from one of the maintainers

File: plugins/cache-ehcache/src/main/java/org/opensearch/cache/EhcacheDiskCacheSettings.java
Comment (sohami) [inline plugins/cache-ehcache/src/main/java/org/opensearch/cache/EhcacheDiskCacheSettings.java:108]: Will this cause any issue with upgrade path where older nodes have the setting as `long` type vs new nodes has it as `ByteSizeValue` type ? Probably not given it is `NodeScope` but wanted to double confirm on this.

Comment (jainankitk) [reply plugins/cache-ehcache/src/main/java/org/opensearch/cache/EhcacheDiskCacheSettings.java:108]: > Will this cause any issue with upgrade path where older nodes have the setting as long type vs new nodes has it as ByteSizeValue type ? Probably not given it is NodeScope but wanted to double confirm on this.

Thinking again on this, it will break for old nodes. Since once the setting is preserved as memorySizeSetting in the cluster state, old nodes will not be able to apply the latest cluster state and cluster manager will kick those nodes out of the cluster.

Comment (peteralfonsi) [reply plugins/cache-ehcache/src/main/java/org/opensearch/cache/EhcacheDiskCacheSettings.java:116]: I've pushed a new change so that there's a new ByteSizeSetting "max_size" rather than "max_size_in_bytes", which has been deprecated. If the new setting is present it uses that value. If not, it looks for the deprecated setting and uses that. If neither are present it uses the default value. 

I'm not totally sure this complexity is worth it, it might just be better to keep the long setting as it was before. Let me know what you think @jainankitk @sohami

**Comment by peteralfonsi** [comment]
Flaky tests: https://github.com/opensearch-project/OpenSearch/issues/14294

**Pull request closed without merging**
